### ISSUE TITLE
add image proxy = get rid of ssl warnings

### ISSUE THREAD

**Issue by ninja-** [RocketChat/Rocket.Chat#971]
should be left as an on/off option if implemented
##

**Comment by Sing-Li** [comment]
@ninja-   Please elaborate.  Aren't these typically interceptors to insert tracker cookies?

**Comment by ninja-** [comment]
@Sing-Li private is one thing, but what I am referring to are the SSL warnings caused by loading images from http links.

**Comment by marceloschmidt** [comment]
Can you suggest how you'd implement this @ninja- ? Thanks

**Comment by ninja-** [comment]
@marceloschmidt there are two ways to do this
one: implement something like /proxy?url=http://image/image.png and use in oembed client
two: let the server fetch & cache images from urls in messages given that /proxy could lead to abuse

**Comment by connorshea** [comment]
You could (and maybe should) block mixed content so the chat stays HTTPS.

**Comment by herbsmn** [comment]
This is definitely something that should be worked on. Seems like using a proxy for images is the way to go, just like how GitHub does it.

**Comment by k0nsl** [comment]
@herbsmn:

Yeah I would love to see this supported as well and I have moderately lobbied the idea.

I suggested to have a look at this project as opposed to camo, partly due to the sheer simplicity of setting it all up:
https://github.com/willnorris/imageproxy

If the developers need a running copy to use for testing, let me know. I have one already running.

**Comment by herbsmn** [comment]
thx for the heads up @k0nsl. so you are already implimenting imageproxy with rocket chat?

**Comment by k0nsl** [comment]
@herbsmn:

No, I haven't. I have just tried to get the ball rolling, so to speak ;)

**Comment by herbsmn** [comment]
@Sing-Li and @marceloschmidt. have either of you looked into this issue recently?

**Comment by herbsmn** [comment]
Is this the same issue as this one? https://github.com/RocketChat/Rocket.Chat/issues/4050

**Comment by childnode** [comment]
this mostly happens when linking external websites not running / using HTTPs link where favicon is automatically sourced via http, so mixed Content warning is build
The proxy might solve at least two, even three problems
1. the image is cached and always visible, even the remote site is down
- the shared content / message is historically stored within the chat platform, even it i
2. the remote server is not polled anytime the chat is being opened
3. the mixed-content security warning is avoided

**Comment by SeanPackham** [comment]
We won't be adding an image proxy, I'd suggest we disable http content from being displayed but if admins allow it, provide a link to open it in another tab.

**Comment by Hudell** [comment]
While we won't be adding an image proxy, a setting to disable the preview of http links is necessary.

**Comment by Yvvw** [comment]
Support for third-party image proxies should really be considered. Image proxies (Internal, or third party) are standard in every competing platform. And critical for user privacy in any situation which allows public access to a chat.

Mattermost implemented third-party proxy support all the way back in early 2018
mattermost/mattermost-server/pull/7991

**Comment by engelgabriel** [comment]
https://github.com/RocketChat/Rocket.Chat/issues/971

**Comment by engelgabriel** [comment]
![image](https://user-images.githubusercontent.com/1000217/51684251-c928d380-1fd2-11e9-9c48-019eefb310af.png)

**Comment by engelgabriel** [comment]
Lets implement support for https://github.com/willnorris/imageproxy

**Comment by engelgabriel** [comment]
Filtering by port has already been done, see the option on the admin:

![image](https://user-images.githubusercontent.com/1000217/51684251-c928d380-1fd2-11e9-9c48-019eefb310af.png)

Lets also implement support for https://github.com/willnorris/imageproxy

**Comment by ThePigsMud** [comment]
Going along with #10606 - if this is an issue to integrate support for signed URLs with that image proxy, could there possibly be an option to simply rewrite the URLs of images to add a proxy URL before them? For example, maybe like `https://open.rocket.chat/proxy/$1`

It seems this would be a good short-term fix and would allow users to use whatever type of image proxy they wanted (possibly even roll your own with nginx!)
